lib/src/connection.dart

```diff
@@ -495,3 +502,76 @@ class Notification {
   /// An optional data payload accompanying this notification.
   final String payload;
 }
+
+class PostgreSQLConnectionPool {
+  PostgreSQLConnectionPool(this.size, this.host, this.port, this.databaseName,
+      {this.username: null,
+        this.password: null,
+        this.timeoutInSeconds: 30,
+        this.timeZone: "UTC",
+        this.useSSL: false});
+
+  bool _isClosed = false;
+  final int size;
+  final String host;
+  final int port;
+  final String databaseName;
+  final String username;
+  final String password;
+  final int timeoutInSeconds;
+  final String timeZone;
+  final bool useSSL;
+  var _connections = new List<PostgreSQLConnection>();
+
+
+  PostgreSQLConnection get connection {
+    if(_connections.length == 0) {
+      throw new PostgreSQLException(
+          "Attempting to get connection, but pool is not open.");
+    }
+    PostgreSQLConnection selectedConnection = _connections.first;
+    int maxLengthQueryQueue() => selectedConnection._queryQueue.length;
+    for(var connection in _connections.skip(1)) {
+      if(connection._queryQueue.length < maxLengthQueryQueue()) {
+        selectedConnection = connection;
+      }
+    }
+    return selectedConnection;
+  }
+
+  Future open() async {
+    if(_isClosed) {
+      throw new PostgreSQLException(
+          "Attempting to reopen a closed connectionPool. Create a new instance instead.");
+    }
+    for(int i = 0; i < size; i++) {
+      _connections.add(_createConnection());
+    }
+    await Future.wait(_connections.map((connection) => connection.open()));
+  }
+
+  PostgreSQLConnection _createConnection() {
+    var connection =new PostgreSQLConnection(host, port, databaseName,
+        username: username,
+        password: password,
+        timeoutInSeconds: timeoutInSeconds,
+        timeZone: timeZone,
+        useSSL: useSSL);
+    connection.done.then((_) {
+      if(_isClosed) {
+        return;
+      }
+      _connections.remove(connection);
+      PostgreSQLConnection newConnection = _createConnection();
+      _connections.add(newConnection);
+      newConnection.open();
```

Comment:
If this fails, the exception will go uncaught; it ought to be caught here. I think you can get away with discarding the error because there are other mechanisms in place, but this needs to be tested considerably.